src/spooled.rs

```diff
@@ -0,0 +1,136 @@
+use std::fs::File;
+use std::io::{self, Read, Write, Seek, SeekFrom, Cursor};
+use file::tempfile;
+
+/// An object that behaves like a regular temporary file, but keeps data in
+/// memory until it reaches a configured size, at which point the data is
+/// written to a temporary file on disk, and further operations use the file
+/// on disk.
+#[derive(Debug)]
+pub struct SpooledTempFile {
+    max_size: usize,
+    cursor: Option<Cursor<Vec<u8>>>,
+    file: Option<File>,
```

Comment:
Let's use an enum here. That is:

```rust
enum SpooledInner {
    InMemory(Cursor),
    File(File),
}
```

That way, we can't *have* any invalid cases. It should get rid of the panics.